ast/base.go

```diff
@@ -20,7 +20,16 @@ import (
 // node is the struct implements node interface except for Accept method.
 // Node implementations should embed it in.
 type node struct {
-	text string
+	text   string
+	offset int
+}
+
```

Comment:
may need a comment for the Exported function.

Reply:
Updated.